### DISCUSSION TITLE
How to extract the Table data which spread across the multiple pages( 1-3) and last page has another Table?

### DISCUSSION THREAD

**Discussion by thangarajdeivasikamani** [pymupdf/PyMuPDF#3557] [Q&A]
Hello Team,

I have the pdf as below
page :1 

 pdf which has 2 columns and 4 rows

 First row content occupies full page. 

  Column Name  | description

  Test1                |   60 to 70 lines of text
         

Page: 2
 Second row occupies fist half of the  page

  Test2                |   20 to 30 lines of text

 Third  row start second half of the page.

  Test3               |   30 to 40 lines of text

Page: 3

 Third rows ends in3rd page

 Test3               |   5 lines

  Fourth rows 

 Test3               |   2 lines

in Page 3 new Table starting as below

5 columns 2 rows

Column Name

                   Column 1  | Column 2 | Column 3 | Column 4 | Column 5

Row 1              1           |   2               |  3            | 4               |              5
Row 2              1 0         |   11             |  13          | 14            |            15

Summary:

Case 1:

Table 1 start from page 1 and End at Page 3

Table 2 start from Page 3 and End at page 3

Case 2:

Table 1 start from page 1 and End at Page 3

Table 2 start from Page 3 and End at page 4

**Comment by JorjMcKie** [comment]
Please be aware that this way of posting questions leads to nothing:
You must provide a PDF or nobody can have any clue what is going on.

**Comment by thangarajdeivasikamani** [comment]
[Test_multipage.pdf](https://github.com/user-attachments/files/15743256/Test_multipage.pdf)


Attached the pdf

**Reply by JorjMcKie** [reply]
Sorry - you did **_not_** attach the PDF!

**Reply by thangarajdeivasikamani** [reply]
Sorry. Pls recheck

**Comment by JorjMcKie** [comment]
I extracted the tables on all pages without issue.
what is your problem?

**Reply by chandan-artpark** [reply]
His problem is not with extracting the tables themselves, but rather understanding how to handle the tables that span across multiple pages. Specifically:

1) How to extract and recombine tables that start on one page and end on another (e.g., Table 1 which spans pages 1 to 3)?
2) How to handle cases where a new table starts on the same page where the previous table ends (e.g., Table 2 starts on page 3 right after Table 1)?

Ps: Even I am dealing with this issue, any insights on this matter is greatly appreciated.

**Reply by JorjMcKie** [reply]
You need to develop your own code to deal with these situations.
The table finder doesn't know better than you whether or not a table on a page continues the last one on a previous page.
Obvious criteria are same column count. If you are lucky, the table header row may have been repeated. But otherwise ...

But **_when_** you determine a continuation, you must use table extractions like `table.extract()[:]` or `table.to_pandas()` to join table content.

**Reply by JorjMcKie** [reply]
There is a table joiner Jupyter notebook [here](https://github.com/pymupdf/PyMuPDF-Utilities/blob/master/table-analysis/join_tables.ipynb).

**Reply by chandan-artpark** [reply]
Thank you for the quick replies, will let you know if I find a good solution to this problem, or if I have any further queries.
